### PR TITLE
Add types declaration file with Diagram class

### LINKED ISSUE #7: TypeScript types declaration file
Hello, I saw your post on Reddit and thought this package could use a types declaration file so that TypeScript developers are able to utilise it when developing a project with this package. Let me know if you'd like one and I can make one for you.

### PR DESCRIPTION
**Pull request by geopic**
Resolve #7 

Let me know if there are any issues :)

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+declare module 'cli-diagram' {
+  class Diagram extends Array {
+    constructor(options?: object);
+
+    draw(): string;
+    height: number;
+    box(contents: string, options?: object): Diagram;
+    container(contents: string, options?: object): Diagram;
+    lines(lines: number | string[], options?: object): Diagram;
+    arrow(lines: number | string[], options?: object): Diagram;
```
Comment (Lakitna) [inline src/index.d.ts]: Arrow is actually `arrow(arrows: string[] | ({direction: string, label: string})[], options?: object): Diagram`

Though I don't expect a lot of people using the latter option.

Comment (geopic) [reply src/index.d.ts]: Sorted. :ok_hand:

**Comment by geopic** [comment]
ESLint is failing the build as it isn't configured to work with TypeScript...

**Comment by Lakitna** [comment]
Updated the eslint config for you :)

Also added `types` to package.json

**Pull request merged**